### PR TITLE
Feature/issue9 add functional test for malformed json

### PR DESCRIPTION
**Pull request by sharonx**
Note that LogEntry_pb2.py is generated from LogEntry.proto using protoc command. Adding it to the repo to avoid additional setup step.

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import time
+from common import start_logging_plugin, \
+    kill_logging_plugin
+
+
+def pytest_addoption(parser):
+    parser.addoption("--splunkd-url",
+                     help="splunkd url used to send test data to. \
+                          Eg: https://localhost:8089",
+                     default="https://52.53.254.149:8089")
```
Comment (luckyj5) [inline test/conftest.py]: Can we please remove the real IP and update the default address with sudo value? 
May be leave default as localhost.

File: test/common.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    '''
+    send a request to the plugin to start logging
+    @param: file_path
+    '''
+    req_obj = {
+        "File": file_path,
+        "Info": {
+            "ContainerID": "test",
+            "Config": {
+                "splunk-url": "https://52.53.254.149:8088",
```
Comment (luckyj5) [inline test/common.py]: Can we please remove the IP and token values and update them with sudo values?

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    `source venv/bin/activate`
+1. Install the dependencies  
+    `pip install -r requirements.txt`  
+2. Start the test with the required options configured  
+    `python -m pytest <options>`  
+
+    **Options are:**  
+    --splunkd-url
+    * Description: splunkd url used to send test data to. Eg: https://localhost:8089  
+    * Default: https://52.53.254.149:8089
```
Comment (luckyj5) [inline test/README.md]: Remove this IP and other creds from readme as well. Thanks!

File: test/common.py (hunk 2 above)
Comment (sharonx) [reply test/common.py]: good call. added to command opts

File: test/malformed_data/test_malformed_events.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+handler = logging.StreamHandler(sys.stdout)
+logger.addHandler(handler)
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    ("", 0),  # should not be sent to splunk
+    (" ", 0),  # should not be sent to splunk
+    ("\xF0\xA4\xAD", 1),  # non utf-8 decodable chars should still make it to splunk
+    ("hello", 1),  # normal string should always to sent to splunk
+])
```
Comment (anushjay) [inline test/malformed_data/test_malformed_events.py]: Should we also add a test for malformed JSON structure with missing quotes or braces? EG: {"test":"value}

**Comment by sharonx** [comment]
While I've been pushing new commits (mostly formatting) in, this PR is ready for review! I will stop pushing new changes to this branch except for review comments. Please review.

**Review by dtregonning — APPROVED** [review]
Thankyou for PR @sharonx 
Tested and Approved.

**Pull request merged**